### ISSUE TITLE
Series Stationarization implementation

### ISSUE THREAD

**Issue by ben10ben** [lucidrains/iTransformer#15]
The original paper uses "Series Stationarization" as proposed in "Non-stationary Transformers: Exploring the Stationarity in Time Series Forecasting". 

It could be easily implemented in the forward function, similar to the iTransformer and the original paper as shown in the code.

    def forecast(self, x_enc, x_mark_enc, x_dec, x_mark_dec):
        means = x_enc.mean(1, keepdim=True).detach()
        x_enc = x_enc - means
        stdev = torch.sqrt(torch.var(x_enc, dim=1, keepdim=True, unbiased=False) + 1e-5)
        x_enc /= stdev

and for rescaling after the attention compute

        dec_out = dec_out * (stdev[:, 0, :].unsqueeze(1).repeat(1, self.pred_len, 1))
        dec_out = dec_out + (means[:, 0, :].unsqueeze(1).repeat(1, self.pred_len, 1))

Am I missing something that would need to be changed as well for this to potentially work as intended?

If you are open to having this as an alternative to revin, I can adapt the code to this implementation and create a pull request.

Thanks for the code, already running some experiments with the model and changing some things and feeling able to adapt it to my needs.

**Comment by lucidrains** [comment]
let me know if [this](https://github.com/lucidrains/iTransformer/commit/5c4f9ac67f7b3872e515fd5febc08311e0da18e6#diff-9a91e17c03b7bfa52dc07f1c6b167e37f396fab11cefcc9e2deeb89e0f7aca49R107) works

**Comment by lucidrains** [comment]
@ben10ben on this subject, i still think the most correct way would be to gather up the statistics, project them into a conditioning vector, and use adaptive normalization throughout the transformer. the other way would be to create [number tokens](https://arxiv.org/abs/2310.02989) and offer them to the transformer on another axis

**Comment by ben10ben** [comment]
yes this works. 

Still trying to wrap my head around it, but very much agree, that normalisation information needs to be propagated to the transformer. To my understanding, RevIn would be an alternative implementation to de-stationary attention? 

This is my big question for the paper if the iTransformer solves this issue somehow. I wanted to try the Stationarization on its own. Maybe some theoretical understanding on my side is still missing.

**Comment by lucidrains** [comment]
@ben10ben now you can run the experiment side by side with the change of a hyperparameter

share your results here!
